### PR TITLE
DrivAerML: 4L/256d+T_max=20 — shorter cosine cycle

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

We have mapped the T_max landscape: T_max=10 gave 15.47%, T_max=50 gave 13.04%, T_max=30 gave 12.96% (best). The optimal T_max appears to be near 30. Is T_max=20 better than T_max=30? We are filling in the gap between the sub-optimal T_max=10 and the current champion T_max=30 to find whether there is a sharper optimum closer to T_max=20.

## Instructions

Run the same command as the DrivAerML baseline but change `--cosine_t_max` to 20:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 20 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-4L256d-tmax
```

**Key change:** `--cosine_t_max 20` (was 30). All other hyperparameters identical to current best.

Report val_primary/surface_rel_l2_pct at every epoch.

## Baseline

- **Dataset:** DrivAerML
- **Primary metric:** `val_primary/surface_rel_l2_pct` (lower is better)
- **Current best:** **12.96%** (val) / 14.41% (test) — PR #2550 (T_max=30)
- **Context:** T_max=10 → 15.47%, T_max=30 → 12.96% (best), T_max=50 → 13.04%
- **Best W&B run:** 8s5i8y06 (violet — 4L/256d, T_max=30, 43 epochs, AdamW lr=5e-4)
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

**CRITICAL flags (required to avoid OOM):**
- `--batch-size 1`
- `--drivaerml-train-surface-points 50000`
- `--drivaerml-eval-surface-points 50000`
- `--max-train-batches 394`
- `--max-eval-batches 200`

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2607 because ADVISOR: Closing — DrivAerML baseline is 5.73% (4L/384d, PR #2602). All 4L/256d experiments are obsolete. You'll receive a new 4L/384d assignment shortly..
